test/ng/templateRequestSpec.js

```diff
@@ -158,6 +158,37 @@ describe('$templateRequest', function() {
     }).not.toThrow();
   }));
 
+  it('should accept empty templates and refuse null or undefined templates in cache',
+    inject(function($rootScope, $templateRequest, $templateCache, $sce) {
+
+    // Will throw on any template not in cache.
+    spyOn($sce, 'getTrustedResourceUrl').and.returnValue(false);
+
+    expect(function() {
+      $templateRequest('tpl.html'); // should go through $sce
+      $rootScope.$digest();
+    }).toThrow();
+
+    $templateCache.put('tpl.html'); // is a no-op, so $sce check as well.
+    expect(function() {
+      $templateRequest('tpl.html');
+      $rootScope.$digest();
+    }).toThrow();
+
+    $templateCache.put('tpl.html', null); // makes no sense, but it's been added, so trust it.
+    expect(function() {
+      $templateRequest('tpl.html');
+      $rootScope.$digest();
+    }).not.toThrow();
+
+    $templateCache.put('tpl.html', ''); // should work (empty template)
```

Comment:
Since you have already called `$templateCache.put()` previously, this test is not so reliable.
I would either empty the cache first ot use the `they` helper to make each case its own testcase.